src/strings/ops.c

```diff
@@ -607,13 +607,25 @@ MVMString * MVM_string_replace(MVMThreadContext *tc, MVMString *original, MVMint
     return result;
 }
 
+static MVMString * string_from_strand_at_index(MVMThreadContext *tc, MVMString *a, MVMuint16 index) {
+    MVMStringStrand *ss = &(a->body.storage.strands[index]);
+    return MVM_string_substring(tc, ss->blob_string, ss->start, ss->end - ss->start);
+}
+
 /* Append one string to another. */
-static MVMint32 final_strand_matches(MVMThreadContext *tc, MVMString *a, MVMString *b) {
+static MVMuint16 final_strand_match_with_repetition_count(MVMThreadContext *tc, MVMString *a, MVMString *b) {
     if (a->body.storage_type == MVM_STRING_STRAND) {
-        MVMStringStrand *ss = &(a->body.storage.strands[a->body.num_strands - 1]);
-        if (ss->end - ss->start == MVM_string_graphs(tc, b))
-            if (MVM_string_equal_at(tc, ss->blob_string, b, ss->start))
+        MVMStringStrand *sa = &(a->body.storage.strands[a->body.num_strands - 1]);
+        if (sa->end - sa->start == MVM_string_graphs(tc, b)) {
+            if (MVM_string_equal_at(tc, sa->blob_string, b, sa->start))
                 return 1;
+        }
+        else if (b->body.storage_type == MVM_STRING_STRAND && b->body.num_strands == 1) {
```

Comment:
And a comment above this conditional as well to describe what it is checking in the 2nd part